datumaro/plugins/data_formats/coco/base.py

```diff
@@ -115,30 +116,43 @@ def __iter__(self):
         yield from self._items.values()
 
     def _load_categories(self, json_data, *, keep_original_ids):
+        self._categories = {}
+
         if has_meta_file(self._rootpath):
             labels = parse_meta_file(self._rootpath).keys()
             self._categories = {AnnotationType.label: LabelCategories.from_iterable(labels)}
             # 0 is reserved for no class
             self._label_map = {i + 1: i for i in range(len(labels))}
-            return
-
-        self._categories = {}
+        else:
+            if self._task in [
+                CocoTask.instances,
+                CocoTask.labels,
+                CocoTask.person_keypoints,
+                CocoTask.stuff,
+                CocoTask.panoptic,
+            ]:
+                self._load_label_categories(
+                    self._parse_field(json_data, "categories", list),
+                    keep_original_ids=keep_original_ids,
+                )
 
-        if self._task in [
-            CocoTask.instances,
-            CocoTask.labels,
-            CocoTask.person_keypoints,
-            CocoTask.stuff,
-            CocoTask.panoptic,
-        ]:
-            self._load_label_categories(
-                self._parse_field(json_data, "categories", list),
-                keep_original_ids=keep_original_ids,
+            if self._task == CocoTask.person_keypoints:
```

Comment:
Here too

Reply:
This `if` is an additional category, so we may not.